Fix reply command

Picred opened the pull request:
### Prerequisites

- [x] I have read and understood the [contributing guide](https://github.com/TendTo/dasaturn/blob/main/.github/CONTRIBUTING.rst).
- [x] The commit message follows the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) guidelines.
- [x] Tested the changes locally with a real Telegram bot.
- [ ] Introduced tests for the changes have been added (for bug fixes / features).
- [ ] Docs have been added/updated (for bug fixes / features).
- [ ] I have updated the `CHANGELOG.rst` file with an overview of the changes made.

### Description

Fix reply command when it is triggered by bot's menu command

### Issue closed by this PR

- [ closes #147 ]

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### Python version you are using

Python 3.10.12

Helias commented on src/spotted/handlers/reply.py:
```diff
@@ -16,7 +16,7 @@ async def reply_cmd(update: Update, context: CallbackContext):
     info = EventInfo.from_message(update, context)
 
-    if len(info.text) <= 7:  # the reply is empty
+    if len(info.text) <= 7 or Config.post_get("channel_tag")[1:] in info.text.lower():  # the reply is empty
```
but if I have
`/reply@Spotted_DMI_Bot blablabla`
this condition will be true but I am using the reply correctly

Picred replied:
I think when you use the full-command `/reply@Spotted_DMI_Bot answer` it is only because you accidentally click the suggested command of the bot menu. Usually you just use `/reply answer` without including `@Spotted_DMI_bot` in the command

Helias replied:
but it's wrong conceptually that you can't use the full-command
you could change the condition into something like
`if len(...) <= SOMETHING:`

where SOMETHING is
- `7` when `not bot_tag in info.text.lower()` 
-  `7 + len(config bot_tag)` when `bot_tag in info.text.lower()`

TendTo commented:
This small change in the _src/spotted/handlers/reply.py_ file should fix the issue:

```py
async def reply_cmd(update: Update, context: CallbackContext):
    info = EventInfo.from_message(update, context)
    ### use the new args property of the info object
    if len(info.args) == 0:  # the reply is empty
        await info.bot.send_message(
            chat_id=info.chat_id,
            text="La reply è vuota\n"
            "Per mandare un messaggio ad un utente, rispondere al suo post o report con /reply "
            "seguito da ciò che gli si vuole dire",
        )
        return
    ### the reply text from the args
    reply_text = " ".join(info.args)
    g_message_id = update.message.reply_to_message.message_id
    if (pending_post := PendingPost.from_group(admin_group_id=info.chat_id, g_message_id=g_message_id)) is not None:
        await info.bot.send_message(
            chat_id=pending_post.user_id,
            text=f"COMUNICAZIONE DEGLI ADMIN SUL TUO ULTIMO POST:\n{reply_text}",
        )
    elif (report := Report.from_group(admin_group_id=info.chat_id, g_message_id=g_message_id)) is not None:
        await info.bot.send_message(
            chat_id=report.user_id, text=f"COMUNICAZIONE DEGLI ADMIN SUL TUO ULTIMO REPORT:\n{reply_text}"
        )
```

It appears that both the command's short and long versions (with the bot tag) behave correctly this way.
Maybe add a couple of tests to make sure the problem won't appear again in the future.

TendTo commented:
And a smaller thing: don't add this fix in the changelog under version 3.0.0 since that version has already been released and cannot be changed.
Add a new version (3.0.1) and the fix under that one. We will accumulate all the changes under that version until it gets released. Then, we increase the version and proceed in the same way for future features or fixes.

Picred commented:
I updated the reply command and removed the previous addition of the entry in the `CHANGELOG.md`. 
I prefer to better understand how it is used before introducing new lines

Unfortunately I don't have time to add the tests, sorry.

Thanks so much @TendTo for the suggestion 😄

TendTo commented:
Just fix the linting issues (there is an unused import and you accidentally removed the function docstring) and we will be golden.
